src/java.base/share/classes/java/net/InetAddress.java

```diff
@@ -657,46 +657,46 @@ public String getCanonicalHostName() {
      */
     private static String getHostFromNameService(InetAddress addr, boolean check) {
         String host = null;
-            try {
-                // first lookup the hostname
-                host = nameService.getHostByAddr(addr.getAddress());
-
-                /* check to see if calling code is allowed to know
-                 * the hostname for this IP address, ie, connect to the host
-                 */
-                if (check) {
-                    @SuppressWarnings("removal")
-                    SecurityManager sec = System.getSecurityManager();
-                    if (sec != null) {
-                        sec.checkConnect(host, -1);
-                    }
+        try {
+            // first lookup the hostname
+            host = nameService.getHostByAddr(addr.getAddress());
+
+            /* check to see if calling code is allowed to know
+             * the hostname for this IP address, ie, connect to the host
+             */
+            if (check) {
+                @SuppressWarnings("removal")
+                SecurityManager sec = System.getSecurityManager();
+                if (sec != null) {
+                    sec.checkConnect(host, -1);
                 }
+            }
 
-                /* now get all the IP addresses for this hostname,
-                 * and make sure one of them matches the original IP
-                 * address. We do this to try and prevent spoofing.
-                 */
+            /* now get all the IP addresses for this hostname,
+             * and make sure one of them matches the original IP
+             * address. We do this to try and prevent spoofing.
+             */
 
-                InetAddress[] arr = InetAddress.getAllByName0(host, check);
-                boolean ok = false;
+            InetAddress[] arr = InetAddress.getAllByName0(host, check);
+            boolean ok = false;
 
-                if(arr != null) {
-                    for(int i = 0; !ok && i < arr.length; i++) {
-                        ok = addr.equals(arr[i]);
-                    }
+            if (arr != null) {
+                for (int i = 0; !ok && i < arr.length; i++) {
+                    ok = addr.equals(arr[i]);
                 }
+            }
 
-                //XXX: if it looks a spoof just return the address?
-                if (!ok) {
-                    host = addr.getHostAddress();
-                    return host;
-                }
-            } catch (SecurityException e) {
-                host = addr.getHostAddress();
-            } catch (UnknownHostException e) {
+            //XXX: if it looks a spoof just return the address?
```

Comment:
looks _like_ a spoof

Reply:
Thanks Brian. Changed it as suggested :)